source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -581,21 +581,23 @@ RedisClusterFactory::createClusterWithConfig(
   if (!cluster.has_cluster_type() || cluster.cluster_type().name() != "envoy.clusters.redis") {
     return absl::InvalidArgumentError("Redis cluster can only created with redis cluster type.");
   }
+  auto resolver =
+      THROW_OR_RETURN_VALUE(selectDnsResolver(cluster, context), Network::DnsResolverSharedPtr);
```

Comment:
nit:  can we do same thing as other clusters here, like:

auto resolver_or_error = selectDnsResolver(cluster, context);
  THROW_IF_NOT_OK(resolver_or_error.status());

Then using std::move(resolver_or_error.value() in line 592.

Reply:
I think `THROW_IF_NOT_OK` is significantly more readable than THROW_OR_RETURN_VALUE, too (i.e. consistency is nice, and my preference is for consistency to be achieved in that direction.)